### PR TITLE
Send connection scores for every track per subscriber

### PR DESCRIPTION
**Pull request by shishirng**
Signed-off-by: shishir gowda <shishir@livekit.io>

### CODE REVIEW

**Comment by frostbyte73** [comment]
Should we be tracking all statistics per track instead of per participant? Or, would it make sense to do something similar to jitter/delay and just take the worst value?

File: pkg/rtc/room.go [hunk 1]
```diff
@@ -760,11 +760,30 @@ func (r *Room) connectionQualityWorker() {
...
 	}
 }
 
+func (r *Room) updateStatsConnectionScores(participants []types.Participant) {
+	for _, p := range participants {
+		for _, track := range p.GetPublishedTracks() {
+			if track.IsMuted() {
+				continue
+			}
+			r.telemetry.UpdateRating(p.ID(), track.ID(), livekit.StreamType_DOWNSTREAM, track.GetConnectionScore())
```
Comment (davidzhao) [inline pkg/rtc/room.go:775]: typo?

```suggestion
			r.telemetry.UpdateRating(p.ID(), track.ID(), livekit.StreamType_UPSTREAM, track.GetConnectionScore())
```

File: pkg/telemetry/telemetryservice.go [hunk 2]
```diff
@@ -20,6 +20,7 @@ type TelemetryService interface {
 	OnDownstreamPacket(participantID string, bytes int)
 	HandleRTCP(streamType livekit.StreamType, participantID string, pkts []rtcp.Packet)
 	Report(ctx context.Context, stats []*livekit.AnalyticsStat)
+	UpdateRating(participantID string, trackSid string, kind livekit.StreamType, rating float64)
```
Comment (davidzhao) [inline pkg/telemetry/telemetryservice.go:23]: is this connection score instead of rating?

File: pkg/rtc/room.go (hunk 1 above)
Comment (frostbyte73) [reply pkg/rtc/room.go:775]: also `DOWNSTREAM` on L782

**Comment by shishirng** [comment]
> Should we be tracking all statistics per track instead of per participant? Or, would it make sense to do something similar to jitter/delay and just take the worst value?

we could always calculate per track score using jitter/delay and packets sent/lost that is already captured. Wanted to capture  the scores that we report to participants to help with debugging

**Comment by frostbyte73** [comment]
> > Should we be tracking all statistics per track instead of per participant? Or, would it make sense to do something similar to jitter/delay and just take the worst value?
> 
> we could always calculate per track score using jitter/delay and packets sent/lost that is already captured. Wanted to capture  the scores that we report to participants to help with debugging

We're intentionally combining all those stats because we wanted to send per participant, but I think our only reasoning was that it was simpler. I think as it stands it becomes a bit confusing, we should either A) add connection quality to `Stats` and have all stats per track, B) combine track connection info and send per participant (sounds like we don't want this), or C) use a different data structure instead of adding track ID and having 2 separate uses for `AnalyticsStat`

**Comment by ashellunts** [comment]
I am working currently on approach A) as agreed on last weekly sync.

**Comment by shishirng** [comment]
@ArthurHlt lets also try to use ConnectionStats instead of stats.buffers too

**Pull request closed without merging**